openshift-knative-operator/pkg/common/api.go

```diff
@@ -0,0 +1,177 @@
+package common
+
+import (
+	"fmt"
+
+	"strings"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/blang/semver/v4"
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"knative.dev/pkg/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// UpgradePodDisruptionBudget upgrade the API version to policy/v1
+func UpgradePodDisruptionBudget() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "PodDisruptionBudget" {
+			return nil
+		}
+		if u.GetAPIVersion() != "policy/v1beta1" {
+			return nil
+		}
+		u.SetAPIVersion("policy/v1")
+		return nil
+	}
+}
+
+// UpgradeHorizontalPodAutoscaler upgrade the API version to autoscaling/v2
+func UpgradeHorizontalPodAutoscaler() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "HorizontalPodAutoscaler" {
+			return nil
+		}
+		if u.GetAPIVersion() != "autoscaling/v2beta2" {
+			return nil
+		}
+		u.SetAPIVersion("autoscaling/v2")
+		return nil
+	}
+}
+
+// SetSecurityContextForAdmissionController set the required pod security context to avoid issues on K8s 1.25+.
+// For more check:  https://connect.redhat.com/en/blog/important-openshift-changes-pod-security-standards
+func SetSecurityContextForAdmissionController() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		switch u.GetKind() {
+		case "Deployment":
```

Comment:
soon we have `statefulsets.apps` here too /cc @pierDipi

Reply:
Yes I agree

Reply:
We would need this, let me know if I should follow up for statefulset supports.